src/_pytest/assertion/rewrite.py

```diff
@@ -199,7 +199,11 @@ def _early_rewrite_bailout(self, name, state):
 
         # For matching the name it must be as if it was a filename.
         parts[-1] = parts[-1] + ".py"
-        fn_pypath = py.path.local(os.path.sep.join(parts))
+        try:
+            fn_pypath = py.path.local(os.path.sep.join(parts))
+        except EnvironmentError:
```

Comment:
I'm not happy with that approach, and I'm open to suggestions.

Reply:
lets use a pathlib path instance or a string instead, i - anything thats not turning itself into a absolute path by force will be good

Reply:
Problem is the fnmatch call we need to perform a little down, see my description on the PR.

Reply:
https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.match

Reply:
Didn't know about that function TBH, but the main point is that  `py.path.local.fmatch` [has its own implementation](https://github.com/pytest-dev/py/blob/796d5a0d92f6eef6f5e28dfffa3a613c73a77b22/py/_path/common.py#L432-L453), which we know from past experience (#2140) that it diverges from the [standard fnmatch module](https://github.com/python/cpython/blob/master/Lib/fnmatch.py#L19) and from [pathlib2.PurePath.match](https://github.com/mcmtroffaes/pathlib2/blob/ac68beb6459f664c748ed62e5746362d35991056/pathlib2/__init__.py#L1138-L1161).

If we use different fnmatch algorithms, I'm afraid we will start to see false positives here (like happened in #2140) and not rewrite some modules.

Reply:
i reviewed pathlib.Path.match - it does it completely correct and better than localpath - as such i consider a port to pathlib required